plugin/completion.py

```diff
@@ -40,11 +41,100 @@
+        detail = item.get('detail')
+        documentation = self.normalized_documentation(item)
+
+        # don't show the detail in the cooperate AC popup,
+        # if it is alread shown in the AC details filed.
+        self.is_detail_shown = bool(detail)
+
+        content = self.get_content(documentation, detail)
+        self.show_popup(content)
+
+        if not detail or not documentation:
+            # To make sure that the details or documentation fields don't exist
+            # we resove the completion item.
+            # If those fields appear after the item is resolved
+            # update the popup.
+            self.do_resolve(item)
+
+    def show_popup(self, content):
+        mdpopups.show_popup(
+            self.view,
+            content,
+            flags=sublime.COOPERATE_WITH_AUTO_COMPLETE,
+            max_width=480,
+            max_height=333,
+            allow_code_wrap=True
+        )
+
+    def update_popup(self, content):
+        mdpopups.update_popup(
+            self.view,
+            content
+        )
+
+    def do_resolve(self, item: dict) -> None:
+        session = session_for_view(self.view, 'completionProvider', self.view.sel()[0].begin())
+        if not session:
+            return
 
-    def run(self, edit: sublime.Edit) -> None:
-        self.view.show_popup('<div style="padding: 10px;">Showing documentation is under construction</div>',
-                             sublime.COOPERATE_WITH_AUTO_COMPLETE)
+        client = client_from_session(session)
+        if not client:
+            return
+
+        completion_provider = session.get_capability('completionProvider')
+        has_resolve_provider = completion_provider and completion_provider.get('resolveProvider', False)
+        if has_resolve_provider:
+            client.send_request(Request.resolveCompletionItem(item), self.handle_resolve_response)
+
+    def handle_resolve_response(self, item):
+        documentation = self.normalized_documentation(item)
+        detail = item.get('detail')
+
+        content = self.get_content(documentation, detail)
+        self.show_popup(content)
+
+    def get_content(self, documentation, detail):
+        content = ""
+        if detail and not self.is_detail_shown:
+            content += """<div class='highlight' style='margin: 10px'>
+                <p>{}</p>
+            </div>""".format(detail)
+
+        if documentation:
+            content += """<div style='margin: 10px'>
+                <p>{}</p>
+            </div>""".format(documentation)
+
+        return content
+
+    def normalized_documentation(self, item: Dict[str, Any]) -> str:
+        lsp_documentation = item.get("documentation")
+        if isinstance(lsp_documentation, str):
+            return mdpopups.md2html(self.view, lsp_documentation)
+        elif isinstance(lsp_documentation, dict):
+            value = lsp_documentation.get("value", '')
+            return mdpopups.md2html(self.view, value)
```

Comment:
You should check the `kind` here:

```
if lsp_documentation.get("kind") == "markdown":
    return mdpopups.md2html(self.view, value)
else:
    # must be plaintext
    return html.escape(value)
```

Reply:
Hmm I agree, 
I should probably fill a but to the css language server. 

I did it this way, becasue the css ls return something like this
```
'documentation': {'value': "Specifies the width of the content area, padding area or border area (depending on 'box-sizing') of certain boxes.\n\nSyntax: <viewport-length>{1,2}\n\nMDN Reference: https://developer.mozilla.org/docs/Web/CSS/width", 'kind': 'plaintext'}
```
note that the `'kind'` is `'plaintext'`. 
But if I parse this `value` as markdown
the link gets clickable:
![image](https://user-images.githubusercontent.com/22029477/81111807-ef4f5a00-8f1d-11ea-9666-4b4c0872bff5.png)

If i just escape it, it won't be clicable.

Reply:
I will change this to follow the spec

Reply:
It's plaintext, but we *could* still make links clickable :) I guess that's a next step after making sure it's plaintext.

Reply:
Or we could ask the vscode-css guys to make their docs `markdown`. Then links will be automatically rendered as clickable.

Reply:
With that example the link will become clickable but also the `<viewport-length>` will disappear as it will be parsed as HTML element. And it's not supposed to be. It's supposed to be shown verbatim.

(CSS server is just stupid. It should use markdown formatting but escape stuff that is supposed to be escaped.)

Reply:
Hmm, now I've looked at your screenshot and it didn't disappear. Hmm. I remember recently someone reported a problem on Discord where an element like that from css server would mess up the text content.

Reply:
I meant this bug https://github.com/sublimelsp/LSP/issues/991